functional/ldap_test.go

```diff
@@ -123,3 +125,52 @@ func TestConnectorLDAPHealthy(t *testing.T) {
 		}
 	}
 }
+
+func TestLDAPPool(t *testing.T) {
```

Comment:
We do both. In my head, the reason for the test is to verify that we do not have lock contention while exercising the various pool functions concurrently. But sure, I'll find a more descriptive name.